lua/nvim-tree/popup-menu/init.lua

```diff
@@ -0,0 +1,79 @@
+local a = vim.api
+
+local M = {}
+
+local function get_node()
+  local get_node_at_cursor = require'nvim-tree.lib'.get_node_at_cursor
+  
+  local current_node = get_node_at_cursor() 
+  return current_node
+end
+
+local current_popup = nil
+
+local winnr
+local bufnr
+
+function get_current_action(_winnr, _bufnr)
+  local node = current_popup.current_node 
+  local cur_pos = a.nvim_win_get_cursor(_winnr)[1]
+  local current_action = a.nvim_buf_get_lines(_bufnr, cur_pos -1, cur_pos, false)[1]
+  
+  for _,v in pairs(require'nvim-tree.actions'.mappings) do
+    if v.action == current_action then
+      require'nvim-tree.actions'.keypress_funcs[current_action](current_popup.current_node)
+    end
+   end
+end
+
+local function setup_mappings()
+  -- TODO: pass mappings as arguments from setup or default
+  a.nvim_buf_set_keymap(bufnr, 'n', '<CR>', string.format([[:lua get_current_action(%s, %s)<CR>]], winnr, bufnr), { noremap = true })
```

Comment:
This one isn't working - it's printing `:lua ...` instead of executing.

Take a look at actions.init.apply_mappings to see how the lua function is mapped.

Reply:
Working nicely now!